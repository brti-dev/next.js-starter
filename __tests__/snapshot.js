import React from 'react'
import renderer from 'react-test-renderer'
import Index from '../pages/index'
import '@testing-library/jest-dom'

test('two plus two is four', () => {
  expect(2 + 2).toBe(4)
})

jest.mock('next/router', () => ({
  useRouter() {
    return {
      route: '/',
      pathname: '',
      query: '',
      asPath: '',
    }
  },
}))

jest.mock('../src/lib/use-media-query', () => {
  return {
    __esModule: true,
    default: jest.fn(() => true),
  }
})

it('renders homepage unchanged', () => {
  const tree = renderer.create(<Index />).toJSON()
  expect(tree).toMatchSnapshot()
})
