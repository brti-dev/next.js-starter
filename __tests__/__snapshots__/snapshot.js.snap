// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders homepage unchanged 1`] = `
Array [
  <a
    data-reach-skip-link=""
    data-reach-skip-nav-link=""
    href="#reach-skip-nav"
  >
    Skip to content
  </a>,
  <header
    className="header"
    id="top"
  >
    <div
      className="heading"
    >
      <h1>
        âœ¨My App
      </h1>
      <button
        className="button color--default shape--circle size--medium variant--default"
        onClick={[Function]}
        style={
          Object {
            "--color": "var(--color-default)",
          }
        }
        type="button"
      >
        <span
          className="main-content"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M4 6h16v2H4zm0 5h16v2H4zm0 5h16v2H4z"
            />
          </svg>
        </span>
      </button>
    </div>
    <nav
      aria-label="Main"
      style={
        Object {
          "--current-index": -1,
          "display": "block",
        }
      }
    >
      <ul>
        <li>
          <a
            href="/"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            Home
          </a>
        </li>
        <li>
          <a
            href="/foo"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            Foo
          </a>
        </li>
        <li>
          <a
            href="/bar"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            Bar
          </a>
        </li>
      </ul>
    </nav>
  </header>,
  <div
    data-reach-skip-nav-content=""
    id="reach-skip-nav"
  />,
  <main
    className="main"
  >
    <main>
      <h2>
        Hello World
      </h2>
      <p>
        Voluptate proident ea aliquip laborum dolor aliqua. Lorem anim commodo cillum et nulla ex labore ea ex reprehenderit voluptate. Eiusmod ex incididunt cillum dolor cupidatat cillum adipisicing.
      </p>
    </main>
  </main>,
  <footer
    className="footer"
  >
    <nav
      aria-label="Footer"
    >
      <ul>
        <li>
          <a
            href="/"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            Home
          </a>
        </li>
        <li>
          <a
            href="/foo"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            Foo
          </a>
        </li>
        <li>
          <a
            href="/bar"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            Bar
          </a>
        </li>
      </ul>
    </nav>
    <button
      className="button color--default shape--circle size--medium variant--default scrollToTop"
      onClick={[Function]}
      style={
        Object {
          "--color": "var(--color-default)",
        }
      }
      type="button"
    >
      <span
        className="main-content"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M6 4h12v2H6zm.707 11.707L11 11.414V20h2v-8.586l4.293 4.293 1.414-1.414L12 7.586l-6.707 6.707z"
          />
        </svg>
      </span>
    </button>
  </footer>,
]
`;
